### PR TITLE
use updater image passed in to the action

### PR DESCRIPTION
**Pull request by jakecoffman**
This allows the API to tell the action which image to use to run the update.

### CODE REVIEW

File: __tests__/api_client.test.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ describe('ApiClient', () => {
     credentialsToken: 'yyy',
     dependabotApiUrl: 'https://localhost',
     dependabotApiDockerUrl: 'https://localhost',
+    updaterImage: '', // irrelevant for this test
```
Comment (mctofu) [inline __tests__/api_client.test.ts:14]: Will it still work if this input isn't provided at all? We'll want to make sure it stays compatible with older GHES versions which won't be populating this.

Comment (jakecoffman) [reply __tests__/api_client.test.ts:14]: Yes, that's done in this or statement: https://github.com/github/dependabot-action/pull/401/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR47-R48

If `params.updaterImage` is falsy (undefined, null, or empty string) then it falls back to the pinned version.

Comment (jakecoffman) [reply __tests__/api_client.test.ts:14]: That link doesn't work unless opened in a new tab, weird! It's this line in main.ts:

```typescript
    // The dynamic workflow can specify which updater image to use. If it doesn't, fall back to the pinned version.
    const updaterImage = params.updaterImage || UPDATER_IMAGE_NAME
```

**Comment by jakecoffman** [comment]
Verified that without the updaterImage parameter it falls back to the pinned SHA: [Slack](https://github.slack.com/archives/CKE86M8V9/p1673374983392569) [Update job](https://github.com/dsp-testing/dependabot-all-updates-test-actions/actions/runs/3886147902/jobs/6630901336#step:6:9)

File: __tests__/api_client.test.ts (hunk 1 above)
Comment (jakecoffman) [reply __tests__/api_client.test.ts:14]: @mctofu I added explicit tests around which version of the updater is used.

**Review by landongrindheim — APPROVED** [review]
🚀

**Pull request merged**